binary/Pos/Binary/Class/TH.hs

```diff
@@ -85,10 +85,10 @@ data Cons = Cons
 
 data Field
     = Field {
-    -- ^ The constructor means that you want
-    -- a field to participate in serialisation/deserialization
+    -- | The constructor means that you want
```

Comment:
Do you know will this produce documentation describing the thing above (`Field` constructor) or field bellow (`fFieldAndType`) ?

I see here https://hackage.haskell.org/package/QuickCheck-2.10.1/docs/src/Test-QuickCheck-Test.html#Args `quickcheck` is also using `-- ^` syntax. So I am not sure is this modification correct?

Reply:
@akegalj The two should be equivalent, according to the documentation, at least: http://haskell-haddock.readthedocs.io/en/latest/markup.html#constructors-and-record-fields

@jcmincke so basically the `-- |` syntax worked but `-- ^` resulted in an error? How weird!

Reply:
As I understand it:

-- | : is used above the field it documents.
-- ^ : directly follows (on the same line? , to check) the field it documents (and is followed by a comma, if needed)

Reply:
Ah, you are probably right. For some reason I prefer the latter, some guys the former. But I appreciate this are pub topics rather than real issues 🍺 😅  (i.e. I'm completely neutral to the use of `--|` here 😉  )